### PR TITLE
MessageBuffer usage improvement

### PR DESCRIPTION
**Pull request by Hadatko**
Signed-off-by: Cervenka Dusan <cervenka@acrios.com>

# Pull request

**Choose Correct**

- [X] bug
- [ ] feature

**Describe the pull request**

https://github.com/EmbeddedRPC/erpc/issues/257
https://github.com/EmbeddedRPC/erpc/issues/256

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

- OS:
- eRPC Version:

**Steps you didn't forgot to do**

- [X] I checked if other PR isn't solving this issue.
- [X] I read Contribution details and did appropriate actions.
- [X] PR code is tested.
- [X] PR code is formatted.

**Additional context**

### CODE REVIEW

**Review by amgross — CHANGES_REQUESTED** [review]

File: erpc_c/infra/erpc_message_buffer.cpp [hunk 1]
```diff
@@ -22,54 +22,61 @@ using namespace std;
...
+        {
+            err = kErpcStatus_MemoryError;
+        }
+        else if ((offset + length) > m_len || (offset + length) <= offset)
+        {
+            err = kErpcStatus_BufferOverrun;
+        }
+        else
         {
             (void)memcpy(m_buf, data, length);
```
[the remark this answers is not in the document]
Comment (Hadatko) [reply erpc_c/infra/erpc_message_buffer.cpp:62]: Agree

**Comment by Hadatko** [comment]
Thank you for PR review. I did it too quickly. 
I accidentaly force pushed (in current job customer is asking less commits more force pushes :D )

**Review by amgross — APPROVED** [review]

**Comment by amgross** [comment]
BTW, I found out I was wrong in comment above and it is OK to check NULL with `if(pointer)`, but still I don't like it and it is better to have uniformity

**Comment by Hadatko** [comment]
> BTW, I found out I was wrong in comment above and it is OK to check NULL with `if(pointer)`, but still I don't like it and it is better to have uniformity

No problem. Misra is also suggesting explicit conditions. Thank you for your engagement.

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
Thank you for the effort!

**Pull request merged**